Journaley/Utilities/PostMarkdownParser.cs

```diff
@@ -0,0 +1,109 @@
+
+            // 0 - stumbles upon the beginning of a <p> tag/usual parsing.
+            // 1 - stumbles upon the end of </p> tag.
+            // 2 - stumbles upon a <pre> tag.
+            var parseState = 0;
+
+            string newLineString;
+            string singleStrike;
+
+            var lines = formattedString.Split(
+                new string[] { "\r\n", "\n" },
+                StringSplitOptions.None);
+
+            foreach (string line in lines)
+            {
+                // We first incrementally check if the line begins with <p> tag. When it does,
+                // We turn on the paragraphBuilder to append the remaining lines.
+
+                // If the checks hit a line with the </p> closing, we disable check and proceed
+                // to parse the text, add <br /> tags and then adding it to builder
+                // where it is assembled with the rest of the file.
+                
+                // We dump the text to builder if it hits a <pre> tag.
+                if (line.Contains("<p>"))
+                {
+                    parseState = 1;
+                }
+                else if (line.Contains("</p>"))
+                {
+                    parseState = 0;
+                }
+                else if (line.Contains("<pre>"))
+                {
+                    parseState = -1;
+                }
+
+                if (parseState == 1)
+                {
+                    paragraphBuilder.AppendLine(line);
+
+                    if (line.Contains("</p>"))
+                    {
+                        // Single line strikethroughs.
+                        singleStrike = Regex.Replace(line, @"(~~)(.*?)(~~)", "<del>$2</del>", RegexOptions.Singleline);
+                        builder.AppendLine(singleStrike);
+                        paragraphBuilder.Clear();
+                        parseState = 0;
+                    }
+                }
+                else if (parseState == 0)
+                {
+                    paragraphBuilder.AppendLine(line);
+
+                    newLineString = paragraphBuilder.ToString();
+
+                    // Multi line strikethrough
+                    newLineString = Regex.Replace(newLineString, @"(~~)(.*?)(~~)", "<del>$2</del>", RegexOptions.Multiline);
+
+                    // Single line breaks append with <br> tags.
+                    newLineString = Regex.Replace(newLineString, @"^([\w\*\>\<\[][^\r\n]*)(?=\r?\n[\w\*\>\<\[].*$)", "$1<br />", RegexOptions.Multiline);
+
+                    builder.AppendLine(newLineString);
+                    paragraphBuilder.Clear();
+                }
+                else
+                {
+                    // For <pre> tagged text, just skip check and append to builder
+                    builder.AppendLine(line);
+                }
+            }
+
+            string strippedOverString = Regex.Replace(builder.ToString(), @"^(\r\s)", string.Empty, RegexOptions.Multiline);
```

Comment:
Remove this code, because it has some unintended behavior. This also removes the whitespaces within the `<pre>` tag, which is definitely not what we want.

Having some whitespaces is not a big problem, and we can just leave them in there. The browser doesn't care about them anyways. Let's just return `builder.ToString()` here.